pkg/storkctl/migration.go

```diff
@@ -330,58 +330,73 @@ func updateCRDObjects(ns string, activate bool, ioStreams genericclioptions.IOSt
 				return
 			}
 			for _, o := range objects.Items {
-				specPath := strings.Split(crd.SuspendOptions.Path, ".")
-				if len(specPath) > 1 {
-					var disableVersion interface{}
-					if crd.SuspendOptions.Type == "bool" {
-						disableVersion = !activate
-					} else if crd.SuspendOptions.Type == "int" {
-						replicas, _ := getSuspendIntOpts(o.GetAnnotations(), activate, ioStreams)
-						disableVersion = replicas
-					} else if crd.SuspendOptions.Type == "string" {
-						suspend, err := getSuspendStringOpts(o.GetAnnotations(), activate, ioStreams)
-						if err != nil {
-							util.CheckErr(err)
+				annotations := o.GetAnnotations()
+				if annotations == nil {
+					printMsg(fmt.Sprintf("Warn: Skipping CR update %s-%s/%s, annotations not found", strings.ToLower(crd.Kind), o.GetNamespace(), o.GetName()), ioStreams.ErrOut)
+					continue
+				}
+				if crd.SuspendOptions.Path != "" {
+					crd.CustomSuspendOptions = append(crd.CustomSuspendOptions, crd.SuspendOptions)
+				}
+				if len(crd.CustomSuspendOptions) == 0 {
+					continue
+				}
+				for _, suspend := range crd.CustomSuspendOptions {
+					specPath := strings.Split(suspend.Path, ".")
+					if len(specPath) > 1 {
+						var disableVersion interface{}
+						if suspend.Type == "bool" {
+							if val, err := strconv.ParseBool(suspend.Value); err != nil {
+								disableVersion = !activate
+							} else {
+								disableVersion = !val
```

Comment:
Shouldn't you be doing a `!val` only if we are activating a migration? If you are deactivating then you should set whatever is provided in suspend.Value.